### PR TITLE
Fix list-resources.sh to work with the GKE provider

### PR DESCRIPTION
**Pull request by roberthbailey**
Also update the grep command to work for versions of grep that don't understand '\s'.

First step towards diagnosing #19779. 

/cc @ihmccreery

### CODE REVIEW

**Comment by ixdy** [comment]
LGTM

**Comment by ixdy** [comment]
I'd say we should also should get rid of [this line](https://github.com/kubernetes/kubernetes/pull/20143/files#diff-4fa29a2c020d02dffb5ccf9a34823b2eR50) so we can avoid the unnecessary `Attempt 1 to list [resource] in GCE`, though I guess that can be a separate quick PR?

File: cluster/gce/list-resources.sh [hunk 1]
```diff
@@ -59,7 +66,7 @@ echo "Project: ${PROJECT}"
 echo "Region: ${REGION}"
 echo "Zone: ${ZONE}"
 echo "Instance prefix: ${INSTANCE_PREFIX:-}"
-echo "Network: ${NETWORK:-}"
+echo "Network: ${NETWORK}"
```
Comment (ixdy) [inline cluster/gce/list-resources.sh]: one more small suggestion: also print out `Provider: ${KUBERNETES_PROVIDER}`?

Comment (roberthbailey) [inline cluster/gce/list-resources.sh]: done.

**Comment by roberthbailey** [comment]
Comments addressed. PTAL.

**Comment by ixdy** [comment]
LGTM, thanks!

**Comment by ixdy** [comment]
https://storage.cloud.google.com/kubernetes-jenkins/pr-logs/pull/20143/kubernetes-pull-build-test-e2e-gce/25287/artifacts/gcp-resources-cluster-up.txt looks better.

**Comment by roberthbailey** [comment]
@ixdy that's a GCE file -- we'll need to wait until this merges to run it on gke-ci before we see the difference.

**Comment by roberthbailey** [comment]
@k8s-oncall Please merge to help with the flaky test surge

**Comment by roberthbailey** [comment]
@k8s-oncall ping for manual merge....

**Comment by fabioy** [comment]
Merging.

**Comment by roberthbailey** [comment]
Thanks!

**Pull request merged**